### ISSUE TITLE
Binaries missing in pre-releases

### ISSUE THREAD

**Issue by panther2** [triplea-game/triplea#5794]
![bin](https://user-images.githubusercontent.com/20812775/71601916-e99fcd00-2b55-11ea-8a95-b4ae855b35ca.PNG)

**Comment by DanVanAtta** [comment]
Failed with:
```
install4j: compilation failed. Reason: java.lang.OutOfMemoryError: Java heap space
```

**Comment by panther2** [comment]
Closed as latest releases have them included.

**Comment by panther2** [comment]
It happened again:

**Comment by panther2** [comment]
@DanVanAtta

**Comment by panther2** [comment]
CC: @RoiEXLab @ron-murhammer @ssoloff

**Comment by RoiEXLab** [comment]
Still the same error unfortunately :/

**Comment by DanVanAtta** [comment]
Until we actually resolve the problem, this seems to be a liability.
The issue is heap space for the install4j process. We do not control that very directly.
It's less than ideal., I don't think there is a readily available fix to be had here.

**Comment by DanVanAtta** [comment]
FWIW, the install4j gradle task was recently update to 8x; Assuming that is okay, perhaps that may have an impact on this issue. Will be interesting if/when how often we see this problem going forward.

**Comment by panther2** [comment]
@DanVanAtta 
I am reopening this, as we see on https://github.com/triplea-game/triplea/releases
that the creation of binaries seems to be a result of a lottery, still.

**Comment by DanVanAtta** [comment]
It looks like this problem is happening more often than not.

It's not a trivial fix as the problem is not really in our control 100%.

We also do not have a lot to go off of to know where this problem is coming from, secondly the iteration time for building installers is very slow, which could make this a painful fix.

The only things I can think of:
- see if we can add flags to increase heap space
- start removing likely suspects
- downgrade install4j version and/or all dependencies from when this problem started happening.

@RoiEXLab any thoughts on this one by any chance?

**Comment by DanVanAtta** [comment]
@RoiEXLab  I wonder a bit about JavaFX dep's, particularly as install4j seemingly has trouble with them. Perhaps this is related to: https://github.com/triplea-game/triplea/issues/5755

**Comment by RoiEXLab** [comment]
> as install4j seemingly has trouble with them.

What do you mean? Could you name an example? I doubt that it has something to do with it. After all RAM doesn't necessarily have anything to do with actual persistant storage space.

But apart from that. no clue. I remember that we had a somewhat related issue quite some time ago, where the PR builder was constantly running out of memory as well. IIRC @ssoloff created a ticket for this which resulted in the variable `_JAVA_OPTIONS` to be set by travis for the gradle daemon. Maybe something similar could help?

**Comment by DanVanAtta** [comment]
Not finding JAVA_OPTS in history or code. It'd be really ideal if we could fix this with adding resources.

>  Could you name an example? 

Travis is reporting 7k warnings for pack200 issues. 

> I doubt that it has something to do with it.

I suspect that could be right, but we don't know what is causing our heap to run out of space. The timing of adding many megabytes, and perhaps if it's not being handled well and we are getting maybe something like file handles being leaked into heap, then it's conceivable we could run out of space and it be exacerbated with additional javaFx class files.

**Comment by ssoloff** [comment]
> IIRC @ssoloff created a ticket for this which resulted in the variable _JAVA_OPTIONS to be set by travis for the gradle daemon.

#2365 links back to all applicable issues and PRs.

**Comment by DanVanAtta** [comment]
Thanks @ssoloff for the reference, that was a help and a similar fix applied in #5897 seems to have worked.

The latest master build with a 3G heap looks to have successfully built installer artifacts. The build process still died though:

```
Triggering github release by pushing tag: 2.0.17868
+cd website
+git update-index --refresh
_maps/dragon-war.html: needs update
```

**Comment by DanVanAtta** [comment]
Along a few fixes, added a setting for the gradle max heap size to be 3G (increased from 2G) when building installers. Since then have not seen an out of heap failure and the latest release has the installers attached to it.
